modules/ROOT/pages/anypoint-mq/3.x/anypoint-mq-ack.adoc

```diff
@@ -0,0 +1,153 @@
+
+=== Automatic Acknowledgment
+
+To acknowledge a received message automatically only if the flow execution completes successfully, set the `acknowledgementMode` to `AUTO` in the Subscriber operation.
+
+If an error occurs during the flow execution, causing its execution to terminate prematurely, the message is NACKed and returns to the queue for redelivery.
+
+=== Manual Acknowledgment
+
+The `MANUAL` acknowledgment mode delegates all responsibility to the app logic to decide when to acknowledge the message, using the ACK or NACK operations.
+
+With this configuration, every message received by either the Subscriber or Consume operations has an `ackToken` value available in the message attributes that identifies this message uniquely for a given connection.
+
+Save this `ackToken` to use later as input to the ACK or NACK operations.
+
+=== Acknowledgment Timeout
+
+When using either `AUTO` or `MANUAL` acknowledgment modes, you can use the `acknowledgementTimeout` parameter to control how long the message remains in-flight waiting to be acknowledged, before automatically returning to the queue.
+
+By default, the connector honors the time defined at the queue level (*Message Lock Default TTL* value) in the Anypoint Platform MQ console.
+
+When using an `acknowledgementTimeout`, consider expected app-time processing, including time for unexpected delays, such as external systems delay and app back-pressure due to high load. For example, if you expect the consumed message to be processed in 10 seconds, set the `acknowledgementTimeout` to a minimum of 15 seconds.
+
+==== Acknowledgment Timeout with AUTO Acknowledge
+
+After a subscriber consumes a message, a timer starts, indicating how long the message has been in-flight for the broker waiting for acknowledgment. If the flow processing does not end with a success or failure before the `acknowledgementTimeout` limit is reached, the message automatically returns to the queue for redelivery.
+
+==== Acknowledgment Timeout with MANUAL Acknowledge
+
+When consuming a message using `MANUAL` acknowledgment mode in the Subscriber or Consume operations, the app must execute either an ACK or NACK operation using the message `ackToken` before the `acknowledgementTimeout` limit is reached. If neither operation executes with the given token before the timeout, the message automatically returns to the queue for redelivery. Any attempt to acknowledge the message after the timeout limit fails due to an expired token.
+
+== Manual Acknowledgment Using the ACK Operation
+
+Acknowledging a message informs the broker that the message has been processed and must be removed from the queue to prevent redelivery.
+
+The acknowledgment token is a unique identifier for the message that you must use when performing the ACK operation. It is available in the `ackToken` field of the message attributes. For example:
```

Comment:
How does this token relate to the previous sentence? Is this token the means by which the broker is "informed"?

Is "when performing the ACK operation" different from what has been described earlier, re. specifying an acknowledgment configuration or mode?

"It is available" sounds like it already exists, but we are implying here that the audience "must use" it.

GLOBALLY omit the fragment "For example:" and end the sentence previous to it with a colon, which serves the same function.